components/pd_client/src/util.rs

```diff
@@ -456,43 +536,84 @@ impl PdConnector {
                 }
             }
         }
-        None
+        Err(box_err!(
+            "failed to connect to {:?}",
+            previous.get_members()
+        ))
     }
 
-    pub async fn connect_member(&self, peer: &Member) -> Option<PdClientStub> {
-        for ep in peer.get_client_urls() {
-            if let Ok((client, _)) = self.connect(ep.as_str()).await {
-                info!("connected to PD leader"; "endpoints" => ep);
-                return Some(client);
+    pub async fn connect_member(
+        &self,
+        peer: &Member,
+    ) -> Result<(Option<(PdClientStub, String)>, bool)> {
+        let mut network_fail_num = 0;
+        let mut has_network_error = false;
+        let client_urls = peer.get_client_urls();
+        for ep in client_urls {
+            match self.connect(ep.as_str()).await {
+                Ok((client, address, _)) => {
+                    info!("connected to PD member"; "endpoints" => ep);
+                    return Ok((Some((client, address)), false));
+                }
+                Err(Error::Grpc(RpcFailure(RpcStatus { status, details: _ }))) => {
+                    if status == RpcStatusCode::UNAVAILABLE
+                        || status == RpcStatusCode::DEADLINE_EXCEEDED
+                    {
+                        network_fail_num += 1;
```

Comment:
should we log the err to find details error?

Reply:
How about add a metrics counter for `network_fail_num`